utils/utils.go

```diff
@@ -153,42 +207,83 @@ func CreateResultFromEndpoint(ep *api.WorkloadEndpoint) (*current.Result, error)
 	return result, nil
 }
 
-func GetIdentifiers(args *skel.CmdArgs) (workloadID string, orchestratorID string, err error) {
+// PopulateEndpointNets takes a WorkloadEndpoint and a CNI Result, extracts IP address and mask
+// and populates that information into the WorkloadEndpoint.
+func PopulateEndpointNets(wep *api.WorkloadEndpoint, result *current.Result) error {
+	if len(result.IPs) == 0 {
+		return errors.New("IPAM plugin did not return any IP addresses")
+	}
+
+	for _, ipNet := range result.IPs {
+		if ipNet.Version == "4" {
+			ipNet.Address.Mask = net.CIDRMask(32, 32)
+		} else {
+			ipNet.Address.Mask = net.CIDRMask(128, 128)
+		}
+
+		wep.Spec.IPNetworks = append(wep.Spec.IPNetworks, ipNet.Address.String())
+	}
+
+	return nil
+}
+
+type EndpointIdentifiers struct {
```

Comment:
Do you really gain anything here by defining your own type?  What's wrong with using the one from libcalico-go?

If this didn't have the WEP name in it then it would basically be the same, and it's easy to calculate the name once you have the WorkloadEndpointIdentifiers struct in your hand.

Reply:
needed one extra field, but it makes sense. I'll just embed the libcalico one